src/type-translator.ts

```diff
@@ -157,6 +160,8 @@ export class TypeTranslator {
   public symbolToString(sym: ts.Symbol): string {
     // This follows getSingleLineStringWriter in the TypeScript compiler.
     let str = '';
+    let alias = this.symbolsToAliasedNames.get(sym);
+    if (alias) return alias;
```

Comment:
This is great.  I think we even may have a test where one case is commented-out because this logic below gets the wrong name.